nodestream_plugin_neptune/neptune_connection.py

```diff
@@ -0,0 +1,115 @@
+import json
+from abc import ABC, abstractmethod
+from logging import getLogger
+
+from aiobotocore.session import get_session
+
+
+class NeptuneConnection(ABC):
+    async def execute(self, query_stmt: str, parameters):
+        response = None
+        async with self._create_boto_client() as client:
+            try:
+                response = await self._execute_query(client,
+                    query_stmt=query_stmt,
+                    parameters=parameters,
+                )
+
+                code = response["ResponseMetadata"]["HTTPStatusCode"]
+                if code != 200:
+                    self.logger.error(
+                        f"Query `{query_stmt}` failed with response:\n{response}"
+                    )
+
+            except Exception as e:
+                self.logger.exception(
+                    f"Unexpected error for query: {query_stmt}.", e, stack_info=True
+                )
+
+        return response
+
+    @abstractmethod
+    def _create_boto_client(self):
+        pass
+
+    @abstractmethod
+    async def _execute_query(self, client, query_stmt: str, parameters: str):
+        pass
+
+
+class NeptuneDBConnection(NeptuneConnection):
```

Comment:
nit: Adding class docuemntation, explaining connection to Neptune vs Analytics and links to the DBs.

Reply:
I've added some class documentation to Neptune Connector as that is the outermost facing class of the plugin.